lms/views/predicates/_lti_launch.py

```diff
@@ -51,6 +43,64 @@ def __call__(self, context, request):
         return has_module_item_configuration == self.value
 
 
+class BlackboardCopied(Base):
+    """
+    Allow invoking an LTI launch view for newly course-copied Blackboard assignments.
+
+    When a user uses Blackboard's Course Copy feature to copy a course that
+    contains Hypothesis assignments, Blackboard copies the Hypothesis
+    assignments into the new course. Blackboard gives new
+    resource_link_id's to the new copies of the assignments and includes
+    the original resource_link_id in a resource_link_id_history launch
+    param.
+
+    Pass blackboard_copied=True to a view's configuration to allow invoking the
+    view only for newly course-copied Blackboard assignments where we don't yet
+    have a ModuleItemConfiguration for the new resource_link_id but we do have
+    a ModuleItemConfiguration for the resource_link_id_history.
+
+    For example:
+
+        @view_config(..., blackboard_copied=True)
+        def blackboard_course_copied_assignment_launch_view(context, request):
+            ...
+
+    resource_link_id_history is a non-standard and undocumented param
+    that's only used by Blackboard as far as we know, but nothing in this
+    code actually prevents us from using resource_link_id_history if
+    another LMS sends it to us.
+    """
+
+    name = "blackboard_copied"
+
+    def __init__(self, value, config):
+        super().__init__(value, config)
+        self.db_configured = DBConfigured(True, config)
```

Comment:
Yeah. I think this is always the case with using composition instead of inheritance: composition creates more boilerplate code as you have to create the composed object in our `__init__()` (or have it passed in to you) and save it on `self`. But I think composition is nonetheless a better pattern: it keeps classes more decoupled, and code easier to understand